manager/integration/tests/test_node.py

```diff
@@ -0,0 +1,375 @@
+import time
+import common
+import pytest
+import os
+import subprocess
+
+from common import clients  # NOQA
+from common import Gi, SIZE
+from common import get_self_host_id
+
+SMALL_DISK_SIZE = (1 * 1024 * 1024)
+DEFAULT_DISK_PATH = '/var/lib/rancher/longhorn/'
+TEST_WRITE_SIZE = 1 * Gi
+TEST_FILE = '/var/lib/rancher/longhorn/test'
+WAIT_FOR_NODE_STATUS = 60
+
+
+def create_disk(client, vol_name, size, node_id):
+    # create a single replica volume and attach it to node
+    volume = create_volume(client, vol_name, size, node_id, 1)
+
+    for replica in volume["replicas"]:
+        id = replica["hostId"]
+        assert id != ""
+        assert replica["running"]
+
+    # prepare the disk in the host filesystem
+    disk_path = common.prepare_host_disk(volume["endpoint"], volume["name"])
+    return disk_path
+
+
+def clean_disk(client, *args):
+    # clean disk
+    for vol_name in args:
+        # umount disk
+        common.umount_host_disk(vol_name)
+        # clean volume
+        clean_volume(client, vol_name)
+
+    volumes = client.list_volume()
```

Comment:
It's weird for the function to ask for the volume names, also check all the volumes at the same time. It can be either, but not both.

For here, I think it doesn't need to check if no volume has been left. `client` finalizer should able to do that.